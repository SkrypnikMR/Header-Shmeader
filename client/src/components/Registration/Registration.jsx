import React from 'react';
import PropTypes from 'prop-types';
import Form from '../UI/Form';
import Input from '../Ui/Input';
import Button from '../UI/Button';
import { regInputs } from '../../contsants/components_consts';

function Registration(props) {
    const { sendRegistrationRequest } = props;
    return (
        <Form>
            <p>Sign Up</p>
            {regInputs.map((input) => {
                return (
                    <Input
                        width="80%"
                        id={input.id}
                        key={input.id}
                        height="100px"
                        inputHeight="50px"
                        borderRadius="5px"
                        label={input.label}
                        value={props[input.id]}
                        onChange={props[`${input.id}Func`]}
                        placeholder={input.placeholder}
                    />
                );
            })}
            <Button id="registration" onClick={sendRegistrationRequest} content="Submit" height="50px" />
        </Form>
    );
}
Registration.propTypes = {
    sendRegistrationRequest: PropTypes.func.isRequired,
};
export default Registration;
