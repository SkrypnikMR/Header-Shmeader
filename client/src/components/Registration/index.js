import { connect } from 'react-redux';
import Registration from './Registration.jsx';
import { setRegistrationValue, sendRegistrationRequest } from '../../Store/registration/actions';
import { registrationStore } from '../../Store/registration/selectors';

const mapStateToProps = state => ({ fields: registrationStore(state) });

const mapDispatchToProps = { setRegistrationValue, sendRegistrationRequest };

export default connect(mapStateToProps, mapDispatchToProps)(Registration);
