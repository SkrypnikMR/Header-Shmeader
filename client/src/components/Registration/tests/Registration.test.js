import React from 'react';
import { shallow } from 'enzyme';
import Registration from '../Registration.jsx';

describe('Registration', () => {
    let props;
    const sendRegistrationRequest = jest.fn();
    const setRegistrationValue = jest.fn();
    const fields = {
        email: '',
        password: '',
        confrim: '',
        firstName: '',
        lastName: '',
    };
    beforeEach(() => {
        props = {
            setRegistrationValue,
            sendRegistrationRequest,
            fields,
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<Registration {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = shallow(<Registration {...props} />);
        expect(component.find('p')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = shallow(<Registration {...props} />);
        expect(component.find('Input')).toHaveLength(5);
    });
    it('should render button', () => {
        const component = shallow(<Registration {...props} />);
        expect(component.find('Button')).toHaveLength(1);
    });
    it('should click on the button', () => {
        const component = shallow(<Registration {...props} />);
        component.find('Button').simulate('click');
        expect(sendRegistrationRequest).toHaveBeenCalledWith();
    });
    it('should click change input', () => {
        const component = shallow(<Registration {...props} />);
        component.find('Input').first().simulate('change', { name: 'email', value: 'emailValue' });
        expect(setRegistrationValue).toHaveBeenCalledWith({ name: 'email', value: 'emailValue' });
    });
});
