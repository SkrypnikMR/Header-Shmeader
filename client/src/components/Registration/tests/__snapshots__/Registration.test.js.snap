// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Should match snapshot 1`] = `"<div class=\\"styled__StRegDiv-dgfw9l-0 egNhFV\\"><div class=\\"styled__StFormDiv-sc-1xa63f-0 eesBZc\\"><p>تحقق في</p><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"email\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">تحقق في</label><input type=\\"text\\" id=\\"email\\" name=\\"email\\" value=\\"\\" placeholder=\\"...أدخل بريدك الإلكتروني\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"password\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">كلمه السر</label><input type=\\"text\\" id=\\"password\\" name=\\"password\\" value=\\"\\" placeholder=\\"...ادخل رقمك السري\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"confirm\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">التأكيد</label><input type=\\"text\\" id=\\"confirm\\" name=\\"confirm\\" placeholder=\\"...أكد رقمك السري\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"firstName\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">Имя</label><input type=\\"text\\" id=\\"firstName\\" name=\\"firstName\\" value=\\"\\" placeholder=\\"Введите своё имя...\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><div width=\\"80%\\" height=\\"100px\\" class=\\"styled__StInputContainer-w3ykst-3 emTECj\\"><label for=\\"lastName\\" class=\\"styled__StLabel-w3ykst-0 gOEYZv\\">Фамилия</label><input type=\\"text\\" id=\\"lastName\\" name=\\"lastName\\" value=\\"\\" placeholder=\\"Введите свою фамилию...\\" class=\\"styled__StInput-w3ykst-1 eAZEdV\\"/></div><button id=\\"registration\\" type=\\"button\\" height=\\"50px\\" class=\\"styled__StButton-sc-1a6hck8-0 biUGaX\\">ليأتي</button><span>مسجل بالفعل؟<a aria-current=\\"page\\" class=\\"active\\" href=\\"/\\">انتقل إلى إذن</a></span></div></div>"`;
